### PR TITLE
Moved the upload to the store client.

### PR DESCRIPTION
**Pull request by elopio**
LP: #1586874

### CODE REVIEW

**Comment by elopio** [comment]
Requires #530

File: .travis.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ env:
   - TEST_SUITE=unit
 before_install:
   - sudo apt-get -qq update
-  - sudo apt-get install -y python3-coverage python3-distutils-extra
+  - sudo apt-get install -y python3-coverage python3-distutils-extra squashfs-tools
```
Comment (sergiusens) [inline .travis.yml]: You probably want this in `debian/control`s `Build-Depends`

Comment (sergiusens) [inline .travis.yml]: Or I don't understand why you need it here if we just use travis for unit tests.

File: snapcraft/_store.py [hunk 2]
```diff
@@ -59,25 +54,42 @@ def logout():
...
+    logger.info('Uploading existing {}.'.format(snap_filename))
 
-        storeapi.upload(snap_filename, snap_name)
+    try:
+        store = storeapi.StoreClient()
+        result = store.upload(snap_filename)
+    except storeapi.InvalidCredentialsError:
+        logger.info('No valid credentials found.'
+                    ' Have you run "snapcraft login"?')
+        return
```
Comment (sergiusens) [inline snapcraft/_store.py]: This should really `raise`, shouldn't it?

File: snapcraft/_store.py [hunk 3]
```diff
@@ -59,25 +54,42 @@ def logout():
...
-    with tempfile.TemporaryDirectory() as temp_dir:
-        subprocess.check_call(
-            ['unsquashfs', '-d', os.path.join(temp_dir, 'squashfs-root'),
-             snap_path, '-e', os.path.join('meta', 'snap.yaml')])
-        with open(os.path.join(
-                temp_dir, 'squashfs-root', 'meta', 'snap.yaml')) as yaml_file:
-            snap_yaml = yaml.load(yaml_file)
+    # Print another newline to make sure the user sees the final result of the
+    # upload (success/failure).
+    logger.info('')
```
Comment (sergiusens) [inline snapcraft/_store.py]: This should be a `print()`, it makes little sense to log an empty line if it is for ui purposes

File: .travis.yml (hunk 1 above)
Comment (elopio) [inline .travis.yml]: that sounds right. I wonder why the autopkgtests haven't failed.

File: snapcraft/_store.py [hunk 4]
```diff
@@ -59,25 +54,42 @@ def logout():
...
 
-    return snap_yaml['name']
+    if success:
+        message = 'Application uploaded successfully'
+        if revision:
+            message += ' (as revision {})'.format(revision)
```
Comment (sergiusens) [inline snapcraft/_store.py]: Why not `'{} (as revision {})'.format(message, revision)` ?

File: snapcraft/_store.py [hunk 5]
```diff
@@ -59,25 +54,42 @@ def logout():
...
+        logger.info(message)
+    else:
+        logger.info('Upload did not complete.')
+
+    if errors:
+        logger.info('Some errors were detected:\n\n%s\n',
```
Comment (sergiusens) [inline snapcraft/_store.py]: I thought we were going to use `format` for everything

File: snapcraft/_store.py [hunk 6]
```diff
@@ -59,25 +54,42 @@ def logout():
...
+                    '\n'.join(str(error) for error in errors))
+
+    if app_url:
+        logger.info('Please check out the application at: %s\n',
```
Comment (sergiusens) [inline snapcraft/_store.py]: same about `format`

File: snapcraft/storeapi/__init__.py [hunk 7]
```diff
@@ -134,6 +137,36 @@ def logout(self):
...
+            raise InvalidCredentialsError()
+        data = _upload.upload_files(snap_filename, self.updown)
+        success = data.get('success', False)
+        if not success:
+            return data
+
+        result = _upload.upload_app(self.sca, snap_name, data)
+        return result
+
+    def _get_name_from_snap_file(self, snap_path):
```
Comment (sergiusens) [inline snapcraft/storeapi/__init__.py]: this doesn't really need to be a class member, right?

File: .travis.yml (hunk 1 above)
Comment (elopio) [inline .travis.yml]: ah, I know. That is installed now by snapd. I added the build-depend.

File: snapcraft/_store.py (hunk 2 above)
Comment (elopio) [inline snapcraft/_store.py]: you are right, good catch. Also it should be error, not info. I made the fix, and added a test for the raise.

File: snapcraft/_store.py [hunk 8]
```diff
@@ -60,9 +60,9 @@ def upload(snap_filename):
         store = storeapi.StoreClient()
         result = store.upload(snap_filename)
     except storeapi.InvalidCredentialsError:
-        logger.info('No valid credentials found.'
-                    ' Have you run "snapcraft login"?')
-        return
+        logger.error('No valid credentials found.'
+                     ' Have you run "snapcraft login"?')
+        raise
```
Comment (sergiusens) [inline snapcraft/_store.py]: I was thinking more of a `raise EnvironmentError('No valid credentials found.'...)` but this works

File: snapcraft/_store.py (hunk 3 above)
Comment (elopio) [inline snapcraft/_store.py]: done.

File: snapcraft/_store.py (hunk 4 above)
Comment (elopio) [inline snapcraft/_store.py]: done.

File: snapcraft/_store.py (hunk 5 above)
Comment (elopio) [inline snapcraft/_store.py]: done.

File: snapcraft/_store.py (hunk 6 above)
Comment (elopio) [inline snapcraft/_store.py]: done.

File: snapcraft/_store.py (hunk 8 above)
Comment (elopio) [inline snapcraft/_store.py]: mmm, with that the error wouldn't be printed in red. But it makes sense, maybe we should review all the execptions that we are throwing back to main.

File: snapcraft/storeapi/__init__.py (hunk 7 above)
Comment (elopio) [inline snapcraft/storeapi/__init__.py]: moved up.

**Pull request merged**